File: 2008.round1/C/C.py
import math

ifile = open('C.t.in', 'r')
ofile = open('C.t.out', 'w')
TC = int(ifile.readline())

origin = math.sqrt(5) + 3

for tc in xrange(1, TC, 1):
    n = int(ifile.readline())
    ans = int(origin ** n) % 100
    ofile.write("Case #%d: %d\n" % (tc, ans))

File: 2008.round1/C/C.cpp
#include<cstdlib>
#include<iostream>
#include<fstream>
#include<cmath>
using namespace std;

int main(int argc, char const *argv[]) {
    freopen("C.t.in", "r", stdin);
    freopen("C.t.out", "w", stdout);

    int TC,N;
    int tc,i;
    long long ans;
    double origin = 3 + sqrt(5);

    scanf("%d", &TC);

    for(tc=1;tc<=TC;tc++) {
        scanf("%d", &N);
        ans = (long long)(pow(origin, N));
        printf("Case #%d: %lld\n", tc, ans % 1000);
    }
    return 0;
}